File: Basics/Kotlin/app/src/main/java/com/suavepirate/kotlinbasics/samples/staticMembers/AdoptionService.kt
package com.suavepirate.kotlinbasics.samples.staticMembers

import com.suavepirate.kotlinbasics.samples.generics.AdoptablePet

class AdoptionService {
    fun instanceAdopt(pet: AdoptablePet) {
        pet.isAdopted = true
    }

    // static
    companion object {
        fun staticAdopt(pet: AdoptablePet) {
            pet.isAdopted = true
        }
    }
}

fun run() {
    AdoptionService.staticAdopt(AdoptablePet(false))
    AdoptionService().instanceAdopt(AdoptablePet(false))
}

File: Basics/C#/CSharp.Basics/Samples/StaticMembers/AdoptionService.cs
using CSharp.Basics.Samples.Generics;
using System;
using System.Collections.Generic;
using System.Text;

namespace CSharp.Basics.Samples.StaticMembers
{
    public class AdoptionService
    {
        public void InstanceAdopt(AdoptablePet pet)
        {
            pet.IsAdopted = true;
        }

        public static void StaticAdopt(AdoptablePet pet)
        {
            pet.IsAdopted = true;
        }
    }

    class Runner
    {
        // ...
        void Run()
        {
            AdoptionService.StaticAdopt(new AdoptablePet
            {
                IsAdopted = false
            });

            new AdoptionService().InstanceAdopt(new AdoptablePet
            {
                IsAdopted = false
            });
        }
    }

}
